src/app.py

```diff
@@ -512,14 +529,17 @@ def job_cancel(job_id: str) -> StatusSchema:
     tags=["stable"],
     dependencies=[Depends(can_list_queries)],
 )
-def job_statuses() -> JobsSchema:
+def job_statuses(user: User = Depends(current_user)) -> JobsSchema:
     """
     Returns statuses of all the jobs in the system. May take some time (> 1s)
     when there are a lot of them.
     """
     jobs = [db.get_job(job) for job in db.get_job_ids()]
     jobs = sorted(jobs, key=lambda j: j.submitted, reverse=True)
     jobs = [j for j in jobs if j.status != "removed"]
+    if "can_list_all_queries" not in get_user_roles(user):
```

Comment:
Should we maybe use some kind of enum for `can_list_all_queries`?

Reply:
Created a follow-up PR for myself #351